### ISSUE TITLE
Extract default values for functional groups in json file

### ISSUE THREAD

**Issue by jhpoelen** [jhpoelen/fb-osmose-bridge#66]
Extract default value for functional group in json file, so that the wizard can read the default and populate the UI accordingly. This way, only the defaults file has to edited to make changes to the pre-defined functional groups.

Only non-empty functional groups for a chosen state, fao or ecosystem will be shown in the wizard.

**Comment by agruss2** [comment]
@jhpoelen 
Please assign folks here; thanks !

**Comment by jhpoelen** [comment]
@agruss2 all wizard issue go to casey, skit. Please feel free to assign them for wizard related issues from now on.

**Comment by FIN-casey** [comment]
@agruss2 Hi, I think we should give this issue a sooner deadline than implementing the [Plus button](https://github.com/jhpoelen/fb-osmose-bridge/issues/43). The filling up of the contents of the table in Step2 is all hard coded (the fields and texts). By giving the wizard default functional groups, the filling up of the table contents should be dynamically. The function that will be created can also be used in creating a new functional group using the plus button.

**Comment by agruss2** [comment]
@FIN-casey I agree; I modified all deadlines accordingly. Many thanks in advance for keeping us regularly posted on progress on this issue.

**Comment by FIN-JBarile** [comment]
@agruss2 For this we need to assign species to functional groups. In issue #65, which you have already closed, its indicated that we will follow the Ecopath functional groups definition. However, you had noted that these are too many and gave an example. Were you supposed to redefine the groupings? Is there a specification somewhere?

Anyway, here's a sample dataset of fishes in the Gulf of Mexico, with corresponding functional groups based on the Ecopath definition. The second sheet summarizes all groupings covered for the area. Please have a look at https://www.dropbox.com/s/f1n1r4468ln6ivh/SampleDataWithFuncGroups.xls?dl=0. 

Please advise, thanks.

**Comment by agruss2** [comment]
@FIN-JBarile @FIN-casey 
(1) I closed Issue #65, because the definition of groupings there is the one we agreed upon. 
The bridge between FishBase/SeaLifeBase and OSMOSE needs to provide a  list of functional groups to the user, based only on habitats, depth ranges, body sizes and high orders (sharks, rays, etc.) .
==> For instance we do not want the FishBase/SeaLifeBase and OSMOSE to define two functional groups called "pelagic-oceanic Rhincodontidae" and "pelagic-oceanic Cetorhinidae", but only one functional group called "pelagic-oceanic sharks".

(2) I had a look at this: look at https://www.dropbox.com/s/f1n1r4468ln6ivh/SampleDataWithFuncGroups.xls?dl=0 
To my mind, the definition of functional groups there is good. However: (i) I will certainly think of some changes in the definition of functional groups, once Casey has coded a first implementation of the groupings and I have been able to test it; and (ii) Currently, the defined functional groups are only fish functional groups, but, at a later point, we will want the bridge between FishBase/SeaLifeBase and OSMOSE to define fish and invertebrate functional groups. 

(3) Therefore, here are my advice:
(i) Casey codes a first implementation of the groupings, based on the exact same logics that resulted in the groupings in https://www.dropbox.com/s/f1n1r4468ln6ivh/SampleDataWithFuncGroups.xls?dl=0 
(ii) I test the first implementation of the groupings
(iii) I provide suggestions of improvements
(iv) Casey codes another implementation of the groupings
and 
(v) We keep exchanging thoughts on this issue until we reach a point where the definition of functional groups by the bridge between FishBase/SeaLifeBase and OSMOSE is fully satisfying. 

Does this sound good to you? 
Please let me know if you have any other questions/comments. Many thanks!

**Comment by FIN-JBarile** [comment]
@agruss2 Thanks for your advice. We shall use the existing grouping meantime then.

**Comment by agruss2** [comment]
@FIN-JBarile Great, thanks !
Once Casey is done with the first implementation of the groupings, please post a comment in this issue, so that I can test this first implementation and provide suggestions of improvements.

**Comment by FIN-casey** [comment]
@agruss2 the first implementation is done. The table in Step2 is now dynamically created based on the user's selection of ecosystem only. Please test the current implementation.

**Comment by agruss2** [comment]
@FIN-casey @FIN-JBarile  Many thanks for your work on this!
I just tried to run a first test. 
Unfortunately, there is a bug preventing me from downloading the "osmose_config.zip" file. 
Here is what is going on: 
Once I have reached the final page "Download configuration files for your OSMOSE model", I click on "Submit". Then, once the UI indicates "Done preparing configuration", a message saying "Click here to download" appears. However, there is no possibility to click on (i.e., interact with) the message saying "Click here to download" to download the zip file on my computer. 
Could you please fix this bug? 
Thanks a lot!

**Comment by FIN-casey** [comment]
Hi @agruss2 , @jhpoelen 

I tested the UI and the download link of the config file. Below is the screenshot of the error I got. 
![image](https://cloud.githubusercontent.com/assets/18322815/23089787/2689bc2c-f5ca-11e6-8904-bac75ddbb1a9.png)

It seems the error is in the web service of the osmose. The wizard got a 503 error code (service unavailable) from the web service.

**Comment by agruss2** [comment]
@FIN-casey Many thanks for the info! 
@jhpoelen Could you please fix this?

**Comment by jhpoelen** [comment]
I was able to reproduce the issue by the following steps:

1. go to http://fin-casey.github.io/wizard.html
2. select "Gulf of Mexico" ecosystem
3. click "proceed"
4. select first group (out of many!) as background group
5. click proceed a couple of times, enter the time step, then submit

The attached file got submitted the server to generate a osmose configuration. The file contained about 1100 species. As the server is crunching to lookup all the associated traits with the many species, the request timed out and a 503 error was produced. Meanwhile, the server is still looking up all the traits as expected.

@agruss2 please confirm that the attached file contains the expected dynamically created functional groups for the gulf of mexico. [configGulfOfMexico.json.txt](https://github.com/jhpoelen/fb-osmose-bridge/files/812471/configGulfOfMexico.json.txt)

I opened separate issues related to error messaging (#102), dealing with big groups (#103) and trouble shooting (#104).

**Comment by agruss2** [comment]
@jhpoelen Many thanks! I examined the "configGulfOfMexico.json.txt" file and confirm that this file contains the expected dynamically created functional groups for the Gulf of Mexico. 
@FIN-casey @FIN-JBarile Please have a look at the present issue, as well as to issues #102, #103, and #104; and coordinate with Jorrit to make sure that all of the problems mentioned above and in the three latest issues are solved as soon as possible. Thank you very much!

**Comment by jhpoelen** [comment]
@agruss2 does that mean we can close this issue?

**Comment by agruss2** [comment]
@jhpoelen I just ran a test and was able to download the zip file. Therefore, we can indeed close the present issue. Awesome work Jorrit, many thanks !

**Comment by FIN-casey** [comment]
Hi @agruss2 , the extraction of default functional groups when the user selected the **Country/FAO combination** is not yet implemented. I will be doing it next since I have now the data for it. 

What do you think of reopening this issue, since there is still implementation left?

**Comment by agruss2** [comment]
@FIN-casey Thanks a lot for having pointed this out. I think that it is better to keep this issue closed; and to open a new one specifically about the problem that remains. I am going to open a new issue now.
